packages/framework/ai-collab/src/test/explicit-strategy/agentEditingReducer.spec.ts

```diff
@@ -1412,3 +1431,732 @@ describe("applyAgentEdit", () => {
+				destination: {
+					type: "objectPlace",
+					target: targetVectorId,
+					place: "after",
+				},
+			};
+			const result = applyAgentEdit(insertEdit, idGenerator, schema.definitions);
+
+			const expectedNewInsertIndex = 1;
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+			const newlyInsertedNode = view.root.rootVectors.at(expectedNewInsertIndex)!;
+
+			const expectedUDiff: InsertDiff = {
+				type: "insert",
+				nodePath: [
+					{
+						shortId: Tree.shortId(newlyInsertedNode),
+						schemaIdentifier: Tree.schema(newlyInsertedNode).identifier,
+						parentField: expectedNewInsertIndex,
+					},
+					{
+						shortId: undefined, // Tree.shortId() would return undefined for the array since it has no sf.identifier field
+						schemaIdentifier: Tree.schema(view.root.rootVectors).identifier,
+						parentField: "rootVectors",
+					},
+					{
+						shortId: Tree.shortId(view.root),
+						schemaIdentifier: Tree.schema(view.root).identifier,
+						parentField: "rootFieldKey",
+					},
+				],
+				aiExplanation: insertEdit.explanation,
+				nodeContent: JSON.parse(JSON.stringify(newlyInsertedNode)) as unknown,
+			};
+			assert.deepEqual(result.diff, expectedUDiff);
+		});
+
+		it("insert non-primitive into array node via ArrayPlace", () => {
+			const { view, schema } = initializeTree();
+			const targetVectorArrayParentId = idGenerator.getId(view.root) ?? fail("ID expected.");
+
+			const insertEdit: Insert = {
+				explanation: "Insert a vector",
+				type: "insert",
+				content: { [typeField]: TestVector.identifier, x: 2, y: 3, z: 4 },
+				destination: {
+					type: "arrayPlace",
+					parentId: targetVectorArrayParentId,
+					field: "rootVectors",
+					location: "start",
+				},
+			};
+			const result = applyAgentEdit(insertEdit, idGenerator, schema.definitions);
+
+			const expectedNewInsertIndex = 0;
+			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+			const newlyInsertedNode = view.root.rootVectors.at(expectedNewInsertIndex)!;
+
+			const expectedUDiff: InsertDiff = {
+				type: "insert",
+				nodePath: [
+					{
+						shortId: Tree.shortId(newlyInsertedNode),
+						schemaIdentifier: Tree.schema(newlyInsertedNode).identifier,
+						parentField: expectedNewInsertIndex,
+					},
+					{
+						shortId: undefined, // Tree.shortId() would return undefined for the array since it has no sf.identifier field
+						schemaIdentifier: Tree.schema(view.root.rootVectors).identifier,
+						parentField: "rootVectors",
+					},
+					{
+						shortId: Tree.shortId(view.root),
+						schemaIdentifier: Tree.schema(view.root).identifier,
+						parentField: "rootFieldKey",
+					},
+				],
+				aiExplanation: insertEdit.explanation,
+				nodeContent: JSON.parse(JSON.stringify(newlyInsertedNode)) as unknown,
+			};
+			assert.deepEqual(result.diff, expectedUDiff);
+		});
+		// it("insert primitive into array node via ObjectPlace
```

Comment:
I noted this in the previous PR: I don't think we should leave these comments like this. I think it would be better to just leave TODO comments describing the test cases we wish to add with notes explaining what is preventing us from adding them.

Alternatively, we could implement the tests and use `it.skip` to prevent them from running. They would still want explanations in this case explaining why they don't work / what work would need to be done to support them..

Reply:
(Same feedback for the other instances of comments like this in the file)